hazelcast/src/main/java/com/hazelcast/concurrent/lock/LockResourceImpl.java

```diff
@@ -101,7 +104,7 @@ boolean extendLeaseTime(String caller, long threadId, long leaseTime) {
         if (!isLockedBy(caller, threadId)) {
             return false;
         }
-
+        this.blockReads = true;
```

Comment:
I think we can make this parameterized, because we are assuming that this method is used only by two phase transactions.